### DISCUSSION TITLE
has anyone succeed in compiling OM on window system? there're errors that need you to figure out

### DISCUSSION THREAD

**Discussion by dollPrincess** [OpenModelica/OpenModelica#13146] [Q&A] [answered]
C:/OpenModelica/OMCompiler/3rdParty is empty, 
but during compilation msys shell seems to need things that should exist in this file folder to complete the configure process, 
because ucrt64.exe reported following errors:
`CMake Error at OMCompiler/Compiler/boot/CMakeLists.txt:20 (message): error: downloading 'https://github.com/OpenModelica/OMBootstrapping/archive/refs/heads/master.tar.gz' failed status_code: 6 status_string: "Could not resolve hostname" log: Could not resolve host: github.com shutting down connection #0`

I am really convinced ah, I am in your official download, but looked all over the documentation did not see about how this empty folder in the compilation process and is the case of reporting errors how to deal with. You only gave a line of what updates need to be described, which who can read ah?
What should I do with this error due to your empty folder? Do I have to kind the folder or what? In the end for window compilation you know someone clear success?

Oh! These're more! So why ucrt64.exe said "does not contain a CMakeLists.txt file"? Codes were downloaded from your contents.
`CMake Error at cmake/omc_utils.cmake:16 (add_subdirectory):
  The source directory

    C:/OpenModelica/OMSimulator

  does not contain a CMakeLists.txt file.
Call Stack (most recent call first):
  CMakeLists.txt:171 (omc_add_subdirectory)`

`CMake Error: File C:/OpenModelica/OMCompiler/Compiler/boot/bomc/tarball-include/OpenModelicaBootstrappingHeader.h does not exist.
CMake Error at OMCompiler/Compiler/boot/CMakeLists.txt:32 (configure_file):
  configure_file Problem configuring file`

These really bring me unconvience, and I really need answers from your guys. Please contact me as soon as possible!!!!!

**Comment by phannebohm** [comment]
AFAIK, people build OMC on windows regularly.

The 3rdParty directory shouldn't be empty. When cloning the repo, did you forget to clone submodules also?

**Reply by dollPrincess** [reply]
> AFAIK, people build OMC on windows regularly.
> 
> The 3rdParty directory shouldn't be empty. When cloning the repo, did you forget to clone submodules also?

Yeah. I tied directly download the .zip as well. But all the same and there are empty file folders without contents. 
But they are all followed by "@......" and they are in blue colour in the website in git hub. 

**So how could I do to solve this? Why this situation happened? Why the folders are followed by "@......" and  are in blue colour?**

![Snipaste_2024-11-11_16-59-27](https://github.com/user-attachments/assets/bed00a96-5369-4741-a202-6a6c97a90adb)

**Reply by phannebohm** [reply, marked as answer]
Yeah, these are [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). If you clone the repo, you can follow our [readme](https://github.com/OpenModelica/OpenModelica?tab=readme-ov-file#working-with-the-repository).

**Reply by dollPrincess** [reply]
> Yeah, these are [git submodules](https://git-scm.com/book/en/v2/Git-Tools-Submodules). If you clone the repo, you can follow our [readme](https://github.com/OpenModelica/OpenModelica?tab=readme-ov-file#working-with-the-repository).

hhhh
OKay, Thanks a lot! Truely helpful.
